### PR TITLE
Telemetry: Set hostname to "Benchmark" on perf runs

### PR DESCRIPTION
**Pull request by NicholasCouri**
## Description

Today the performance runs set @fluidframework/mocha-test-setup to the hostname telemetry field.  Setting it to Benchmark instead to make it easier to separate our results from the Stress tests and mocha test runs.

### CODE REVIEW

**Review by alexvy86 — COMMENTED** [review]
I can see how this works... but I feel like we're abusing the existing field. I also don't like that this isn't as automatic; as proved by the PR itself, when we create loggers in other places we need to make sure they specify extra property with this "magical value" `"Benchmark"` (in a field called "hostname" that seems very unintuitive); things don't "just work" in all cases. I'd like to discuss this a bit more.

**Comment by NicholasCouri** [comment]
yes, I completely understand and agree with you. Not sure how to better address it.
Here is the task:

Task 4207: Telemetry PerfBenchmark: Data_hostname set to "benchmark" and fix clientType - Boards (azure.com)<https://dev.azure.com/fluidframework/internal/_workitems/edit/4207>


________________________________

**Review by alexvy86 — APPROVED** [review]
Feels much better 😄. Can we just add a bit to mocha-test-setup's README, explaining the new functionality?

I'm still not super happy about using hostname as we do, but approving to move this forward and get a feel for ways of doing it while we settle on a final approach.

Dumping some initial thoughts here:

The way I see it, we'd first have a clear distinction between "operational Fluid telemetry logged while doing something (most probably running tests)" and "synthetic data generated by some process (not necessarily tests)". While the second could arguably be also called telemetry, I'd avoid it to have a clear distinction. Bundle size measurements fall in the second category for me. They're not a result of running tests, and they're not operational Fluid telemetry; we put them in the same table and under the same-ish schema kind of due to inertia, but I think we should move away from that. I'd propose a new table (brainstorming: office_fluid_ffautomation_metrics) to store timeseries-like data, which today is mostly the things we want to track per CI build (perf benchmarks, bundle sizes, etc) separately from the operational telemetry generated by our code.

File: tools/pipelines/test-perf-benchmarks.yml [hunk 1]
```diff
@@ -487,6 +487,9 @@ stages:
               workingDirectory: ${{ variables.testWorkspace }}/node_modules/${{ variables.testPackage }}
               script: |
 
+                FLUID_LOGGER_PROPS=$FLUID_LOGGER_PROPS;
+                echo "FLUID_LOGGER_PROPS = $FLUID_LOGGER_PROPS"
```
[the remark this answers is not in the document]
Comment (NicholasCouri) [reply tools/pipelines/test-perf-benchmarks.yml:491]: Wait, echo would not set the variable, would it ?

**Comment by NicholasCouri** [comment]
> Feels much better 😄. Can we just add a bit to mocha-test-setup's README, explaining the new functionality?
> 
> I'm still not super happy about using hostname as we do, but approving to move this forward and get a feel for ways of doing it while we settle on a final approach.
> 
> Dumping some initial thoughts here:
> 
> The way I see it, we'd first have a clear distinction between "operational Fluid telemetry logged while doing something (most probably running tests)" and "synthetic data generated by some process (not necessarily tests)". While the second could arguably be also called telemetry, I'd avoid it to have a clear distinction. Bundle size measurements fall in the second category for me. They're not a result of running tests, and they're not operational Fluid telemetry; we put them in the same table and under the same-ish schema kind of due to inertia, but I think we should move away from that. I'd propose a new table (brainstorming: office_fluid_ffautomation_metrics) to store timeseries-like data, which today is mostly the things we want to track per CI build (perf benchmarks, bundle sizes, etc) separately from the operational telemetry generated by our code.

I like the idea of separating them. Let's do it :)

**Pull request merged**